### PR TITLE
feat: Add securityContext. Fixes #96

### LINKED ISSUE #96: [argo] Add securityContext to ui and controller
Would be nice to remove as much privileges, force readonly FS, run as non root and so on.

### PR DESCRIPTION
**Pull request by davidkarlsen**
Signed-off-by: David J. M. Karlsen <david@davidkarlsen.com>

Adds security-context.
I wonder if it makes sense to reuse some other charts for dex / redis to avoid reinventing the wheel for these?

Checklist:

* [X] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [X] I have signed the CLA and the build is green.
* [X] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

---
This change is [Reviewable](https://reviewable.io/reviews/argoproj/argo-helm/185)

### CODE REVIEW

**Comment by davidkarlsen** [comment]
@seanson PTAL

File: charts/argo-cd/templates/argocd-application-controller/deployment.yaml [hunk 1]
```diff
@@ -35,6 +35,9 @@ spec:
 {{- toYaml .Values.controller.podLabels | nindent 8 }}
         {{- end }}
     spec:
+      {{- if .Values.global.securityContext.enabled }}
```
Comment (seanson) [inline charts/argo-cd/templates/argocd-application-controller/deployment.yaml]: Switch this to `.Values.global.securityContext` to pair with requested Values change below.

File: charts/argo-cd/values.yaml [hunk 2]
```diff
@@ -12,6 +12,11 @@ global:
     repository: argoproj/argocd
     tag: v1.3.4
     imagePullPolicy: IfNotPresent
+  securityContext:
```
Comment (seanson) [inline charts/argo-cd/values.yaml]: This shouldn't need an enabled, just default `securityContext` to `{}`, removed `enabled` and comment out the below `runAsUser`, etc.

File: charts/argo-cd/templates/redis/deployment.yaml [hunk 3]
```diff
@@ -25,6 +25,9 @@ spec:
         app.kubernetes.io/component: {{ .Values.redis.name }}
     spec:
       automountServiceAccountToken: false
+      {{- if .Values.global.securityContext }}
+      securityContext: {{- toYaml .Values.global.securityContext | nindent 8 }}
+      {{- end }}
```
Comment (louisblin) [inline charts/argo-cd/templates/redis/deployment.yaml]: Hi all,
That PR didn't add a similar entry for `templates/dex/deployment.yaml` with a `global.securityContext`. Was that simply an omission that we should fix?

Comment (davidkarlsen) [reply charts/argo-cd/templates/redis/deployment.yaml]: I guess it was just forgotten - does it work with a similar security-context? I can add a PR.

Comment (louisblin) [reply charts/argo-cd/templates/redis/deployment.yaml]: Yes, I think so. I've been using the same lines in `dex/deployment.yaml` as what you had here for `redis/deployment.yaml` and it's been working fine. Thanks for the PR.

Comment (davidkarlsen) [reply charts/argo-cd/templates/redis/deployment.yaml]: https://github.com/argoproj/argo-helm/pull/248

**Pull request merged**